File: member-service/src/main/java/com/hedvig/memberservice/events/FraudulentStatusUpdatedEvent.java
package com.hedvig.memberservice.events;

import com.hedvig.memberservice.aggregates.FraudulentStatus;
import lombok.Value;

import java.util.HashMap;
import java.util.Map;

@Value
public class FraudulentStatusUpdatedEvent implements Traceable {
  private Long memberId;
  private FraudulentStatus fraudulentStatus;
  private String fraudulentDescription;

  @Override
  public Map<String, Object> getValues() {
    Map result = new HashMap();
    result.put("Fraudulent status", fraudulentStatus);
    result.put("Fraudulent description", fraudulentDescription);
    return result;
  }
}

File: member-service/src/main/java/com/hedvig/memberservice/events/FraudulentStatusUpdatedEvent.kt
package com.hedvig.memberservice.events

import com.hedvig.memberservice.aggregates.FraudulentStatus
import java.util.HashMap

class FraudulentStatusUpdatedEvent(
    override val memberId: Long,
    val fraudulentStatus: FraudulentStatus? = null,
    val fraudulentDescription: String? = null
) : Traceable {
    override fun getValues() = mapOf(
        "Fraudulent status" to fraudulentStatus,
        "Fraudulent description" to fraudulentDescription
    )
}
